.gitignore

```diff
@@ -4,10 +4,14 @@ demos/bitcast/decode
 demos/bitcast/transfers
 dist/
 journal
-main
 simulator
+main
 tigerbeetle
 cluster_*_replica_*.tigerbeetle
 zig/
 zig-cache/
 zig-out/
+
+# Project Files
+.idea
+*.iml
```

Comment:
These editor specific files should really go in your personal global gitignore, not the one for the project. You can run for example
```
git config --global core.excludesFile ~/.gitignore
```
and then add these two rules to `~/.gitignore`.

The same goes for the `.vscode` folder currently present here and for the `.DS_Store` in the viewstamped-replication-made-famous repository

Reply:
Agreed. Thanks. Updated `.gitignore`

Reply:
Am I allowed to resolve this myself? or do I leave the resolution up to the creator to resolve? 
@ifreund @jorangreef

Reply:
Resolving these discussions yourself is fine if you believe you've addressed the problem and no further discussion is needed.

Reply:
Thanks @koekiebox and @ifreund for the great review. Agreed also on the personal global `.gitignore`, I was wanting to suggest that too and my bad for the `.DS_Store` that crept in, we can remove it here too now.

We want to keep the root of the repo, including the `.gitignore`, as clean as possible from artifacts or other environments creeping in. @koekiebox it would be great to see also if we can't move our Docker related files to our `scripts` folder to clean up the root. You should run this by @wilsonianb who set it up.